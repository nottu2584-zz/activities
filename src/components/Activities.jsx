import { default as React } from "react";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  activities: {
    background: "#F3F4F6",
    color: "#282c34",
    borderRadius: 5,
    fontSize: "0.9em",
    overflow: "hidden",
    padding: 0,
  },
  row: {
    display: "flex",
    flexWrap: "wrap",
    width: "100%",
    "&.head": {
      borderBottom: "1px solid #DCDFE5",
      fontFamily: "'Rubik', sans-serif",
      fontSize: "1.2em",
      fontWeight: "bold",
      textTransform: "uppercase",
    },
    "&:nth-child(even)": {
      backgroundColor: "#DCDFE5",
    },
  },
  cell: {
    boxSizing: "border-box",
    flexGrow: "1",
    padding: "0.8em 1.2em",
    overflow: "hidden",
    listStyle: "none",
    "&.col-1": {
      flex: "0 0 50px",
    },
    "&.col-2": {
      flex: "0 0 calc(50% - 50px)",
      textAlign: "left",
    },
    "&.col-3": {
      flex: "0 0 25%",
    },
    "&.col-4": {
      flex: "0 0 25%",
    },
  },
});

const Activities = (props) => {
  const classes = useStyles();
  const { activities, checked, handleOnChange } = props;

  return (
    <div className={`activities ${classes.activities}`}>
      <div className={`${classes.row} row head`}>
        <div className={`${classes.cell} col col-1`}></div>
        <div className={`${classes.cell} col col-2`}>Activity Name</div>
        <div className={`${classes.cell} col col-3`}>COVID Friendly</div>
        <div className={`${classes.cell} col col-4`}>Times Performed</div>
      </div>
      {activities
        ? activities.map((activity, index) => {
            return (
              <div key={index} className={`${classes.row} row row-${index}`}>
                <div className={`${classes.cell} col col-1`}>
                  <input
                    name={`activity-${activity.id}`}
                    id={`activity-${activity.id}`}
                    type="checkbox"
                    checked={checked.some(
                      (item) => item.id === activity.id
                    )}
                    onChange={(e) => handleOnChange(e, activity)}
                  />
                </div>
                <div className={`${classes.cell} col col-2`}>
                  {activity.name}
                </div>
                <div className={`${classes.cell} col col-3`}>
                  {activity.friendly ? "Yes" : "No"}
                </div>
                <div className={`${classes.cell} col col-4`}>
                  {activity.performed}
                </div>
              </div>
            );
          })
        : null}
    </div>
  );
};

export default Activities;
