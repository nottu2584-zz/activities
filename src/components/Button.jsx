
import { default as React } from "react";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  button: {
    background: "#009b72",
    boxShadow: "0 1px 2px #333",
    border: "none",
    borderRadius: "5px",
    color: "#ffffff",
    cursor: "pointer",
    fontFamily: "'Rubik', sans-serif",
    fontSize: "1.1em",
    lineHeight: "1.1em",
    outline: "none",
    padding: "1em 1.25em",
    MozTransition: "box-shadow 0.25s",
    OTransition: "box-shadow 0.25s",
    WebkitTransition: "box-shadow 0.25s",
    transition: "box-shadow 0.25s",
    WebkitFontSmoothing: "antialiased",
    MozOsxFontSmoothing: "grayscale",
  },
  "&:hover": { boxShadow: "0 1px 20px #333" },
});

const Button = (props) => {
  const classes = useStyles();
  
  const {handleOnClick} = props;

  return (
    <button className={classes.button} onClick={(e) => handleOnClick(e)}>
      {props.children}
    </button>
  );
};

export default Button;