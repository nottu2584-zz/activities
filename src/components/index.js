export { default as Activities } from "./Activities";
export { default as Button } from "./Button";
export { default as Modal } from "./Modal";

