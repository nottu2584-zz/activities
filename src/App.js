import { default as React, useEffect, useReducer, useState } from "react";
import { Activities, Button, Modal } from "./components";
import activityReducer from "./reducers/activity";
import ActivityService from "./services/activity.service";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  app: {
    color: "white",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    textAlign: "center",
    "& button": {
      marginTop: "1.5em"
    }
  },
  content: {
    padding: "5%",
    minWidth: "60%",
  },
});

const App = () => {
  const classes = useStyles();

  const initialState = {
    activities: [],
    checked: [],
  };

  const [friendly, setFriendly] = useState(true);
  const [state, dispatch] = useReducer(activityReducer, initialState);

  useEffect(() => {
    ActivityService.getActivities()
      .then((res) => {
        dispatch({ type: "SET_ACTIVITY", payload: res.data.activities });
      })
      .catch((err) => console.log("Error fetching activities", err));
  }, []);

  const handleOnChange = (e, activity) => {
    if (e.target.checked) {
      dispatch({ type: "ADD_ACTIVITY", payload: activity });
    } else {
      dispatch({ type: "REMOVE_ACTIVITY", payload: activity });
    }
  };

  const handleModal = (e) => {
    if (!friendly) setFriendly(true);
  };

  const performActivities = (e) => {
    if (!friendlyActivities()) setFriendly(false);
    else {
      dispatch({ type: "PERFORM_ACTIVITY" });
      dispatch({ type: "UPDATE_ACTIVITY" });
    }
  };

  const friendlyActivities = () => {
    if (state.checked.length > 0) {
      for (let i = 0; i < state.checked.length; i++) {
        if (!state.checked[i].friendly) return false;
      }
    }

    return true;
  };

  return (
    <div className={`app ${classes.app}`}>
      <div className={`content ${classes.content}`}>
        <header className="header">
          <h1>My 2020 Activity Tracker</h1>
        </header>
        <Activities
          activities={state.activities}
          checked={state.checked}
          handleOnChange={handleOnChange}
        ></Activities>
        {state.checked.length ? (
          <Button handleOnClick={performActivities}>Perform Activities</Button>
        ) : null}
      </div>
      {!friendly ? (
        <Modal
          activities={
            state.checked.length
              ? state.checked.filter((activity) => activity.friendly === false)
              : null
          }
          handleModal={handleModal}
          title={`We are sorry`}
          message={`The following activities cannot be done under the current
            circunstances:`}
        ></Modal>
      ) : null}
    </div>
  );
};

export default App;
