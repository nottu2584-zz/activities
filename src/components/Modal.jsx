import { createUseStyles } from "react-jss";
import Button from "./Button";

const useStyles = createUseStyles({
  modal: {
    backgroundColor: "rgba(0, 0, 0, 0.85)",
    position: "fixed",
    height: "100%",
    width: "100%",
    bottom: 0,
    left: 0,
    right: 0,
    top: 0,
    zIndex: 1,
  },
  wrapper: {
    display: "flex",
    flexDirection: "column",
    flexWrap: "wrap",
    alignItems: "center",
    height: "100%",
    justifyContent: "center",
    zIndex: 2,
  },
  content: {
    display: "inline-flex",
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "center",
    maxWidth: "30%",
    textAlign: "center",
    "& p": {
      margin: "1em 0 0",
    },
    "& ul": {
      textAlign: "left",
      margin: "1em 0",
    },
  },
  fullWidth: {
    flex: "1 1 100%",
  },
  button: {
    background: "#009B72",
    boxShadow: "0 1px 2px #333",
    border: "none",
    borderRadius: 5,
    color: "#FFFFFF",
    cursor: "pointer",
    fontSize: "1.25em",
    lineHeight: "1.1em",
    outline: "none",
    padding: "1em 1.25em",
    "-moz-transition": "box-shadow 0.25s",
    "-o-transition": "box-shadow 0.25s",
    "-webkit-transition": "box-shadow 0.25s",
    transition: "box-shadow 0.25s",
    "&:hover": {
      boxShadow: "0 1px 20px #333",
    },
  },
});

const Modal = (props) => {
  const classes = useStyles();

  const { activities, handleModal, message, title } = props;

  return (
    <div className={`modal ${classes.modal}`}>
      <div className={`wrapper ${classes.wrapper}`}>
        <div className={`content ${classes.content}`}>
          <h2 className={classes.fullWidth}>{title}</h2>
          <p className={classes.fullWidth}>{message}</p>
          <ul className={classes.fullWidth}>
            {activities
              ? activities.map((activity, index) => (
                  <li key={index}>{activity.name}</li>
                ))
              : null}
          </ul>
          <Button handleOnClick={handleModal}>Oh, I see!</Button>
        </div>
      </div>
    </div>
  );
};

export default Modal;
