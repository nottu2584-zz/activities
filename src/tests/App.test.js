
import { shallow } from "enzyme";
import { default as React } from "react";
import App from "../App";
import { Activities } from "../components";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useLayoutEffect: jest.requireActual("react").useEffect,
}));

describe("<App />", () => {
  const container = shallow(<App />);
  it("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  it("should have an some wrappers", () => {
    expect(container.find(".app").length).toEqual(1);
    expect(container.find(".content").length).toEqual(1);
  });

  it("should have an <Activities /> component", () => {
    expect(container.find(Activities).length).toEqual(1);
  });
});
