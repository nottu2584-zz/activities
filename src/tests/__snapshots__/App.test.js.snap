// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match the snapshot 1`] = `"<div class=\\"app app-0-2-1\\"><div class=\\"content content-0-2-2\\"><header class=\\"header\\"><h1>My 2020 Activity Tracker</h1></header><div class=\\"activities activities-0-2-3\\"><div class=\\"row-0-2-4 row head\\"><div class=\\"cell-0-2-5 col col-1\\"></div><div class=\\"cell-0-2-5 col col-2\\">Activity Name</div><div class=\\"cell-0-2-5 col col-3\\">COVID Friendly</div><div class=\\"cell-0-2-5 col col-4\\">Times Performed</div></div></div></div></div>"`;
